app/src/main/java/com/framgia/soundcloud/data/source/local/sharedpref/SharedPrefsImplement.java

```diff
@@ -0,0 +1,56 @@
+package com.framgia.soundcloud.data.source.local.sharedpref;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+/**
+ * Created by anh on 05/11/2017.
+ */
+
+public class SharedPrefsImplement implements SharedPrefApi {
+    private static final String PREFS_NAME = "SoundCouldSharedPreferences";
+    private SharedPreferences mSharedPreferences;
+
+    public SharedPrefsImplement(Context context) {
+        mSharedPreferences = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <T> T get(String key, Class<T> clazz) {
+        if (clazz == String.class) {
+            return (T) mSharedPreferences.getString(key, "");
+        } else if (clazz == Boolean.class) {
+            return (T) Boolean.valueOf(mSharedPreferences.getBoolean(key, false));
+        } else if (clazz == Float.class) {
+            return (T) Float.valueOf(mSharedPreferences.getFloat(key, 0));
+        } else if (clazz == Integer.class) {
+            return (T) Integer.valueOf(mSharedPreferences.getInt(key, 0));
+        } else if (clazz == Long.class) {
+            return (T) Long.valueOf(mSharedPreferences.getLong(key, 0));
+        }
+        return null;
+    }
+
+    @Override
+    public <T> void put(String key, T data) {
+        SharedPreferences.Editor editor = mSharedPreferences.edit();
+        if (data instanceof String) {
+            editor.putString(key, (String) data);
+        } else if (data instanceof Boolean) {
+            editor.putBoolean(key, (Boolean) data);
+        } else if (data instanceof Float) {
+            editor.putFloat(key, (Float) data);
+        } else if (data instanceof Integer) {
+            editor.putInt(key, (Integer) data);
+        } else if (data instanceof Long) {
+            editor.putLong(key, (Long) data);
+        }
+        editor.apply();
+    }
+
+    @Override
+    public void clear() {
+        mSharedPreferences.edit().clear();
```

Comment:
WARNING: [CommitPrefEdits] `SharedPreferences.edit()` without a corresponding `commit()` or `apply()` call
 After calling `edit()` on a `SharedPreference`, you must call `commit()` or `apply()` on the editor to save the results.